operator/apis/platform/v1alpha1/securedcluster_types.go

```diff
@@ -197,6 +197,8 @@ type CollectionMethod string
 const (
 	// CollectionEBPF means: use EBPF collection.
 	CollectionEBPF CollectionMethod = "EBPF"
+	// CollectionModernBPF means: use modern_bpf collection.
+	CollectionModernBpf CollectionMehtod = "modern_bpf"
```

Comment:
`BPF` in caps?

Reply:
BTW I'm not very fond of using adjectives such as `new`, `modern`, `ng`, because they tend to look weird after a few months/years at best, and very confusing at worst, when the next shiny hotness is born. Perhaps just call it `BTF` if that's what you mean? :thinking:

Reply:
BTF itself is also not quite right, as it's still a BPF probe, just of different sort -- so it's technically wrong to call it BTF. But we're open for suggestions, theoretically instead of an exact name it could be a "nick name" for the new collection method. Any other ideas?

Reply:
I see. What's different about it? Worst case, perhaps `EBPF2` or pick a cool-sounding name from your favourite sci-fi universe? :wink:

Reply:
@erthalion How does BTF differentiate from the other collection methods? Is there something the consumer of the API should know about this collection method?

Reply:
> What's different about it?

> How does BTF differentiate from the other collection methods? Is there something the consumer of the API should know about this collection method?

The intrinsic difference is about using "modern" (hence the first ideas for the name) kernel functionality, to allow the probe work on any sufficiently new kernel version. The user -visible API is the same, in theory the consumer shouldn't notice any difference whatsoever on supported platforms. So, what's your favorite sci-fi universe then? :)

Reply:
> > What's different about it?
> 
> > How does BTF differentiate from the other collection methods? Is there something the consumer of the API should know about this collection method?
> 
> The intrinsic difference is about using "modern" (hence the first ideas for the name) kernel functionality,

Does this functionality have any feature apart from being modern? :-D

> So, what's your favorite sci-fi universe then? :)

Mine is Lem's but I'm not the implementer here :-)

Reply:
> Does this functionality have any feature apart from being modern? :-D

Yes, it works on *all* new kernels, where the previous was working only on *one* kernel. Maybe a "universal_bpf" then?

> > So, what's your favorite sci-fi universe then? :)
>
> Mine is Lem's but I'm not the implementer here :-)

I would love to see a collection method "ijon_tichy"!

Reply:
Should a user ever prefer EBPF over BPF if their Kernel is on a minimum version? What is the minimum version?

Reply:
This would be described in the documentation, but the short summary is that the new collection method would be preferable when it become stable enough. The minimal supported kernel version is a fluid thing in this case, and Collector would be responsible to figure out if the new probe is supported. If not it will fallback to the regular ebpf.